src/boundaries/stagger_and_step.jl

```diff
@@ -0,0 +1,259 @@
+
+    progress = ProgressMeter.Progress(
+        N; desc = "Saddle estimation: ", dt = 1.0
+    )
+    xi = stagger_trajectory(ds, x0, Tm, isinside, :unif, δ₀, 
+                            γ, max_steps, max_escape_time, rng) 
+    if isnothing(xi)
+        error("Cannot find a stagger trajectory. Choose a different starting point or 
+                search radius δ₀.")
+    end
+
+    v = Vector{typeof(current_state(ds))}(undef, N)
+    v[1] = xi
+    for n in 1:N
+        show_progress && ProgressMeter.update!(progress, n)
+        if escape_time!(ds, xi, isinside, max_escape_time) > Tm
+            reinit!(ds, xi)
+        else
+            xp, Tp = stagger!(ds, xi, Tm, isinside, stagger_mode, δ, γ, max_steps,  
+                              max_escape_time, rng)
+            # The stagger step may fail. We reinitiate the algorithm from a new initial condition.
+            if Tp < 0
+                xp = stagger_trajectory(ds, x0, Tm, isinside, :exp, δ₀, γ, max_steps,  
+                                        max_escape_time, rng) 
+                if isnothing(xp)
+                    error("Cannot find a stagger trajectory. Choose a different starting 
+                          point or search radius δ₀.")
+                end
+                δ = 0.1
+            end
+            reinit!(ds, xp)
+        end 
+        step!(ds)
+        xi = copy(current_state(ds))
+        v[n] = xi
+    end
+    return StateSpaceSet(v)
+end
+
+
+"""
+    stagger_trajectory(ds, x0, Tm, isinside, δ, γ, stagger_mode,
+        max_steps, max_escape_time, rng) -> xi
+
+On success, the function returns a point `xi` with the property `T(xi) > 
+Tm` with a random walk search around the initial coordinates 
+`x0`. `T(xi)` is the escape time of the initial condition `xi` from 
+the bounding box defined by the function `isinside`.  
+In the case where the algorithm cannot find a suitable point, the algorithm 
+returns nothing. 
+
+This is an auxiliary function for [`stagger_and_step`](@ref). 
+Arguments and definitions are identical for both functions. 
+"""
+function stagger_trajectory(ds, x0, Tm, isinside, stagger_mode, δ, γ,
+        max_steps, max_escape_time, rng)
+    T = escape_time!(ds, x0, isinside, max_escape_time)
+    xi = copy(x0) 
+    while !(T > Tm)  # we must have T > Tm at each step 
+        xi, T = stagger!(ds, xi, T, isinside, stagger_mode, δ, γ, max_steps, max_escape_time, rng)
+        if T < 0
+            return nothing
+        end 
+            
+    end
+    return xi
+end
+    
+
+
+function escape_time!(ds, x0, isinside, max_escape_time) 
+    set_state!(ds,x0)
+    reinit!(ds, x0)
+    k = 1; 
+    while isinside(current_state(ds)) 
+        if k > max_escape_time 
+            error("The trajectory did not escape for ", k, " steps, you probably have \
+an attractor in the defined region. Last point evaluated: ", current_state(ds))
+        end
+        step!(ds)
+        k += 1
+    end
+    return current_time(ds)
```

Comment:
```suggestion
    return current_time(ds) - initial_time(ds)
```

This is correct for systems whose initial time happens to not be 0